Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -2025,172 +2120,203 @@ private void UpdateRigHandles()
-            {
-                newState = HandleProximityState.CloseProximity;
-            }
-            else if (closestDistance < handleMediumProximity)
-            {
-                newState = HandleProximityState.MediumProximity;
-            }
-            else//far
+            //only use proximity effect if nothing is being dragged or grabbed
+            if (currentPointer == null)
             {
-                newState = HandleProximityState.FullsizeNoProximity;
-            }
-
-            ScaleHandle(handle, handleVisual, newState, handleSize);
-
-            if (renderer != null && state != newState)
-            {
-                renderer.material = newState == HandleProximityState.CloseProximity ? handleGrabbedMaterial : handleMaterial;
-            }
-            return newState;
-        }
-
-        private void ScaleHandle(Transform handleRoot, Transform handleVisual, HandleProximityState targetState, float handleSize)
-        {
-            float targetScale = targetState == HandleProximityState.CloseProximity ? closeScale : (targetState == HandleProximityState.MediumProximity ? mediumScale : farScale);
-            float weight = targetState == HandleProximityState.CloseProximity ? closeGrowRate : (targetState == HandleProximityState.MediumProximity ? mediumGrowRate : farGrowRate);
-            float newLocalScale = (handleVisual.transform.localScale.x * (1.0f - weight)) + (handleSize * targetScale * weight);
-            handleVisual.transform.localScale = new Vector3(newLocalScale, newLocalScale, newLocalScale);
-        }
+                proximityPointers.Clear();
+                proximityPoints.Clear();
 
-        private void HandleProximityScaling(bool forceFar = false)
-        {
-            if (corners?.Count > 0 || balls?.Count > 0)
-            {
-                inputPoints.Clear();
-                //only use proximity effect if nothing is being dragged or grabbed
-                if (currentPointer == null && forceFar == false)
+                // Find all valid pointers
+                foreach (var inputSource in CoreServices.InputSystem.DetectedInputSources)
                 {
-                    if (TryGetPointerPoint(Handedness.Left, out Vector3 point))
+                    foreach (var pointer in inputSource.Pointers)
                     {
-                        inputPoints.Add(point);
+                        if (pointer.IsInteractionEnabled && !proximityPointers.Contains(pointer))
+                        {
+                            proximityPointers.Add(pointer);
+                        }
                     }
-                    if (TryGetPointerPoint(Handedness.Right, out point))
+                }
+
+                // Get the max radius possible of our current bounds plus the proximity
+                float maxRadius = Mathf.Max(Mathf.Max(currentBoundsExtents.x, currentBoundsExtents.y), currentBoundsExtents.z);
+                maxRadius *= maxRadius;
```

Comment:
Hm....why are we squaring maxRadius here?

Reply:
Because the comparison later is against sqrMagnitude. Sqrt is very expensive in distance